_rules/table-headers-attribute-refer-to-data-cells-a25f45.md

```diff
@@ -0,0 +1,318 @@
+
+```html
+<table>
+	<tr>
+		<th id="name" colspan="2">Name</th>
+	</tr>
+	<tr>
+		<th headers="name">Firsname</th>
+		<th headers="name">Lastname</th>
+	</tr>
+</table>
+```
+
+### Failed
+
+#### Failed Example 1
+
+The `headers` attribute on the cell refers to an element outside the same `table`.
+
+```html
+<span id="elmOutsideTable">Project Costs</span>
+
+<table>
+	<tr>
+		<th id="headerOfColumn">Projects</th>
+	</tr>
+	<tr>
+		<td headers="elmOutsideTable">15%</td>
+	</tr>
+</table>
+```
+
+#### Failed Example 2
+
+The `td` element has a `headers` attribute referring to an ID that does not exist within the same `table` (or within the [document](https://dom.spec.whatwg.org/#concept-document)). Even if some of the ID referred to exist, a single non-existing one is enough to fail the rule.
+
+```html
+<table>
+	<tr>
+		<th colspan="2" id="header1">Projects</th>
+	</tr>
+	<tr>
+		<th id="e1" headers="header1">1</th>
+		<th id="e2" headers="header1">2</th>
+	</tr>
+	<tr>
+		<td colspan="2" headers="header1 e1 NOT-EXIST">15%</td>
+	</tr>
+</table>
+```
+
+#### Failed Example 3
+
+Both `td` elements have `headers` attribute which refers to ID that do not exist within the same `table`.
+
+```html
+<table>
+	<tr>
+		<th id="header1">Projects</th>
+		<th id="header2">Exams</th>
+	</tr>
+	<tr>
+		<td headers="NOT-EXIST-1">15%</td>
+		<td headers="NOT-EXIST-2">15%</td>
+	</tr>
+</table>
+```
+
+#### Failed Example 4
+
+The `headers` attribute on the cell refers to an element inside the same `table` which does not have a role of `rowheader` or `columnheader`.
+
+```html
+<table>
+	<tr>
+		<td>
+			<span id="headerProject">Projects</span>
+		</td>
+	</tr>
+	<tr>
+		<td headers="headerProject">
+			15%
+		</td>
+	</tr>
+</table>
+```
+
+### Inapplicable
+
+#### Inapplicable Example 1
+
+There is no `headers` attribute.
+
+```html
+<table>
+	<tr>
+		<th scope="col">Projects</th>
+		<th scope="col">Exams</th>
+	</tr>
+	<tr>
+		<td>15%</td>
+		<td>45%</td>
+	</tr>
+</table>
+```
+
+#### Inapplicable Example 2
+
+The `table` has a `role="presentation"` and thus is not [included in the accessibility tree](#included-in-the-accessibility-tree).
+
+```html
+<table role="presentation">
+	<tr>
+		<td id="header1">Project Status</td>
+	</tr>
+	<tr>
+		<td>15%</td>
+	</tr>
+</table>
+```
+
+#### Inapplicable Example 3
+
+The `table` is not [visible](#visible).
+
+```html
+<table style="display: none;">
+	<tr>
+		<td id="header1">Project Status</td>
+	</tr>
+	<tr>
+		<td>15%</td>
+	</tr>
+</table>
+```
+
+#### Inapplicable Example 4
+
+The rule applies only to `headers` attribute within a `table` element.
+
+```html
+<div role="table">
+	<div role="row">
+		<div role="columnheader" id="header1">Projects</div>
+		<div role="columnheader" id="header2">Exams</div>
+	</div>
+	<div role="row">
+		<div role="cell" headers="header2">15%</div>
+		<div role="cell" headers="header1">15%</div>
+	</div>
+</div>
+```
+
+#### Inapplicable Example 5
```

Comment:
Essentially the same as Inapplicable example 2, and we're not going to list all the ways a table can be excluded from the accessibility tree. Can be removed.